pykern/pkio.py

```diff
@@ -18,11 +18,26 @@
 import re
 import shutil
 import six
+import tempfile
 
 #: used during unit testing see ``pykern.pkunit.save_chdir``
 pkunit_prefix = None
 
 
+def atomic_write(filename, content, mode='w'):
+    """Writes to file atomically
+
+    Args:
+        filename (str or py.path.Local): File to write to
+    """
+    with tempfile.NamedTemporaryFile(
+        mode=mode, dir=os.path.dirname(str(filename))) as f:
```

Comment:
Once converted to a py.path, you can then change the suffix to pkio-tmp, and it'll be good enough, the new code would then look like:

```py
n = p.new(ext='pkio-tmp')
n.write(contents)
n.rename(p)
```